src/main/java/com/yscope/logging/log4j1/DateConversionPatternExtractor.java

```diff
@@ -0,0 +1,134 @@
+package com.yscope.logging.log4j1;
+
+import java.util.ArrayList;
+
+/**
+ * A class to extract all date conversion patterns from a Log4j conversion
+ * pattern. This is useful for Log4j appenders that want to store a log event's
+ * message separate from its timestamp.
+ */
+class DateConversionPatternExtractor {
+  private enum STATES {
+    LITERAL, PERCENT, DATE_SPECIFIER, DATE_SPECIFIER_OPEN_BRACE, DATE_SPECIFIER_OPEN_QUOTE,
+  }
+
+  private final ArrayList<SubstringBounds> dateConversionPatterns = new ArrayList<>();
+  private final String conversionPatternWithoutDates;
+
+  public DateConversionPatternExtractor (String conversionPattern) {
```

Comment:
Appreciate if any examples added to the comment!

Reply:
Added an example.